### PR TITLE
Don't prepend marker for deserialization of root struct

### PR DESCRIPTION
**Pull request by Boog900**
I didn't like the old method, this one is better

### CODE REVIEW

File: src/de.rs [hunk 1]
```diff
@@ -262,8 +266,13 @@ impl<'de, 'a, 'b> serde::Deserializer<'de> for &'a mut Deserializer<'b> {
...
     {
-        let marker = self.read_marker()?;
-        self.dispatch_based_on_marker(marker, visitor)
+        if !self.read_header {
+            self.read_header = true;
+            visitor.visit_map(MapAccess::with_varint_encoded_fields(self)?)
+        } else {
+            let marker = self.read_marker()?;
+            self.dispatch_based_on_marker(marker, visitor)
+        }
```
Comment (thomaseizinger) [inline src/de.rs:275]: Nit: Can we use early-return please.

```suggestion
            return visitor.visit_map(MapAccess::with_varint_encoded_fields(self)?)
        }
        
        let marker = self.read_marker()?;
        self.dispatch_based_on_marker(marker, visitor)
```

**Review by thomaseizinger — COMMENTED** [review]
It seems like we'll need to bump the MSRV in CI. Can you add a changelog entry for that please?

**Comment by thomaseizinger** [comment]
> I didn't like the old method, this one is better

To be honest, I am not sure I fully agree. I see it as a short-coming of the data format that the root struct does not have a marker. Hence, I find it quite easy to understand that we "fix" this short-coming by prepending the marker.

I am not opposed to changing it, just wondering why you think this approach is better.

**Comment by Boog900** [comment]
I agree this is a short coming of the data format but the old method felt too much like a hack for my liking. Plus doing what it was doing before causes the data to be cloned before de-serializing which will reduce performance, there probably is a way to do it without cloning but we may as well just do this.

**Pull request merged**